src/main/java/io/gatling/mojo/AbstractEnterprisePluginMojo.java

```diff
@@ -103,21 +95,33 @@ private EnterpriseClient initEnterpriseClient() throws MojoFailureException {
                   "MY_API_TOKEN_VALUE");
       throw new MojoFailureException(msg);
     }
-    final String pluginTitle = getClass().getPackage().getImplementationTitle();
-    final String pluginVersion = getClass().getPackage().getImplementationVersion();
-    if (pluginTitle == null || pluginVersion == null) {
-      // Should no happen if the plugin is built and packaged properly
-      throw new IllegalStateException("Gatling plugin title and version not found");
+    return new PluginConfiguration(
+        enterpriseUrl,
+        apiToken,
+        controlPlaneUrl,
+        BuildTool.MAVEN,
+        pluginVersion(),
+        forceBatchMode,
```

Comment:
Warning (I should've seen that in the commons PR) : we never have a concept named "batch", it's specific to Maven. We talk about interactive or headless (or non-interactive sometimes).

It's not a blocker, but not future-proof.